pkg/cluster/docker_desktop.go

```diff
@@ -111,20 +111,42 @@ func (c DockerDesktopClient) Quit(ctx context.Context) error {
 
 func (c DockerDesktopClient) ResetCluster(ctx context.Context) error {
 	klog.V(7).Infof("POST /kubernetes/reset\n")
+
+	checkErrors := func(err error, statusCode int) error {
```

Comment:
tbh, i think the `checkErrors` function here is trying to be too cute about reducing duplication at the expense of another level of indirection/complexity that's going to lead to bugs (like the one i pointed out below)